ownership/sync-repos.py

```diff
@@ -0,0 +1,255 @@
+"""
+Synchronize ownership information between repositories and spreadsheet.
+
+Finds mismatches between the repositories listed in the ownership
+tracking spreadsheet and the actual list of repositories in a
+collection of Github organizations.
+"""
+
+import click
+import csv
+import github3
+from github3.exceptions import NotFoundError
+from itertools import groupby
+import re
+
+def bail(message):
+    raise KnownError(message)
+
+class KnownError(Exception):
+    def __init__(self, message):
+        super().__init__(message)
+        self.message = message
+
+@click.command()
+@click.option(
+    '--github-username',
+    help="Username corresponding to the Github access token"
+)
+@click.option(
+    '--token-file',
+    help=("File containing Github token with repo scope on applicable orgs.\n"
+          "(Recommend using a Bash process substitution.)")
+)
+@click.option(
+    '--spreadsheet-csv',
+    help="Ownership spreadsheet CSV data (for local testing)"
+)
+def main(github_username, token_file, spreadsheet_csv):
+    orgs = ['edx', 'edx-solutions', 'edx-ops']
+
+    with open(token_file, 'r') as tf:
+        token = tf.readline().rstrip('\n')
+    hub = github3.login(github_username, token)
+
+    try:
+        repos_gh = list(fetch_github(hub, orgs))
+        repos_ss = list(fetch_spreadsheet(spreadsheet_csv, orgs))
+
+        run_compare(hub, orgs, repos_gh, repos_ss)
+    except KnownError as e:
+        print(e.message)
+
+
+def run_compare(hub, orgs, repos_gh, repos_ss):
+    # List of actions to recommend at end of run.
+    #
+    # This list will *grow* as the comparison progresses.
+    actions = []
```

Comment:
Can you also document that each action is a dict and what the keys in that dict mean here?